Sanitize error when trying to run GUI without XIndex: lshw-02.14/src/gui/integration/console.apps
===================================================================
--- a/src/gui/integration/console.apps
+++ b/src/gui/integration/console.apps
@@ -1,3 +1,3 @@
 USER=root
-PROGRAM=/usr/sbin/gtk-lshw
+PROGRAM=/usr/sbin/lshw-gtk
 SESSION=true
--- a/src/gui/integration/gtk-lshw.desktop
+++ b/src/gui/integration/gtk-lshw.desktop
@@ -3,7 +3,7 @@
 Comment=HardWare LiSter
 Categories=Application;System;X-Red-Hat-Base;X-Fedora;
 Icon=/usr/share/lshw/artwork/logo.svg
-Exec=/usr/bin/gtk-lshw
+Exec=/usr/bin/lshw-gtk
 Type=Application
 Terminal=false
 Encoding=UTF-8
--- a/src/lshw.cc
+++ b/src/lshw.cc
@@ -11,6 +11,7 @@
 #include <string.h>
 #include <stdlib.h>
 #include <iostream>
+#include <errno.h>
 
 #ifndef NONLS
 #include <locale.h>
@@ -30,8 +31,8 @@
   fprintf(stderr, _("\t-xml            output hardware tree as XML\n"));
   fprintf(stderr, _("\t-short          output hardware paths\n"));
   fprintf(stderr, _("\t-businfo        output bus information\n"));
-  if(getenv("DISPLAY") && exists(SBINDIR"/gtk-lshw"))
-    fprintf(stderr, _("\t-X              use graphical interface\n"));
+  if(getenv("DISPLAY") && exists(SBINDIR"/lshw-gtk"))
+      fprintf(stderr, _("\t-X              use graphical interface\n"));
   fprintf(stderr, _("\noptions can be\n"));
 #ifdef SQLITE
   fprintf(stderr, _("\t-dump OUTFILE   save hardware tree to a file\n"));
@@ -211,7 +212,19 @@
     exit(1);
   }
 
-  if(enabled("output:X")) execl(SBINDIR"/gtk-lshw", SBINDIR"/gtk-lshw", NULL);
+/*  if(enabled("output:X")) execl(SBINDIR"/gtk-lshw", SBINDIR"/gtk-lshw", NULL); */
+
+  if(enabled("output:X")) {
+        execl(SBINDIR"/lshw-gtk", SBINDIR"/lshw-gtk", NULL);                                                                                                                                                    
+        // Execl only returns if an error has occurred.
+        if ( errno == ENOENT ) {
+             fprintf(stderr, "ERROR: Sorry, cannot run the X11/GTK interface because %s/lshw-gtk\n is not available.\n", SBINDIR);                                                                               
+             fprintf(stderr, "HINT: Install the lshw-gtk package in Debian.\n");                                                                                                                                 
+        } else {
+             fprintf(stderr, "ERROR: There was an error when trying to execute %s/lshw-gtk: %s\n", SBINDIR, strerror(errno));                                                                                    
+             fprintf(stderr, "Will just print the information on screen\n");                                                                                                                                     
+        }
+  }
 
   if (geteuid() != 0)
   {
